src/allocatorStrategies/GaugeDistributionStrategy.sol

```diff
@@ -0,0 +1,565 @@
+        return _campaign.lastProcessedEpoch > 0 ? _campaign.lastProcessedEpoch + 1 : _campaign.startEpoch;
+    }
+
+    /// @notice Gets the effective end epoch for a campaign
+    /// @param _campaign Campaign storage reference
+    /// @param _currentEpoch Current epoch
+    /// @return The effective end epoch (considering continuous campaigns)
+    function _getEffectiveEndEpoch(
+        GaugeDistributionCampaign storage _campaign,
+        uint256 _currentEpoch
+    )
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 endEpoch = _campaign.endEpoch == 0 ? _currentEpoch : _campaign.endEpoch;
+        return endEpoch > _currentEpoch ? _currentEpoch : endEpoch;
+    }
+
+    // =========================================================================
+    // External Functions
+    // =========================================================================
+
+    /// @inheritdoc IAllocatorStrategy
+    /// @dev Creates a multi-epoch campaign with start and end epochs
+    function setAllocationCampaign(uint256 _campaignId, bytes calldata _auxData) public override onlyAuthorized {
+        // Campaign shouldn't already exist
+        if (campaigns[_campaignId].startEpoch != 0) {
+            revert CampaignAlreadyExists(_campaignId);
+        }
+
+        // Decode epoch bounds
+        (uint256 startEpoch, uint256 endEpoch) = abi.decode(_auxData, (uint256, uint256));
+
+        // Validate epoch bounds
+        if (startEpoch == 0) revert InvalidEpochBounds(startEpoch, endEpoch);
+        if (endEpoch != 0 && startEpoch > endEpoch) revert InvalidEpochBounds(startEpoch, endEpoch);
+
+        // Create new campaign
+        GaugeDistributionCampaign storage campaign = campaigns[_campaignId];
+        campaign.startEpoch = startEpoch;
+        campaign.endEpoch = endEpoch;
+
+        emit AllocationCampaignCreated(plugin, _campaignId);
+    }
+
+    /// @notice Sets the distribution amount for a specific epoch
+    /// @param _campaignId Campaign identifier
+    /// @param _epochId Epoch to set distribution for
+    /// @param _amount Amount to distribute for this epoch
+    function setEpochDistribution(uint256 _campaignId, uint256 _epochId, uint256 _amount) external onlyAuthorized {
+        GaugeDistributionCampaign storage campaign = campaigns[_campaignId];
+        uint256 currentEpoch = _getCurrentEpoch();
+
+        // Validate params
+        {
+            if (_amount == 0) revert InvalidDistributionAmount();
+
+            // Validate campaign exists
+            _validateCampaignExists(_campaignId);
+
+            // Validate epoch is within campaign bounds
+            _validateEpochInCampaign(campaign, _epochId);
+
+            // Validate that if epoch is in the past, it has a snapshot
+            _validatePastEpochHasSnapshot(_epochId, currentEpoch);
+        }
+
+        _setEpochDistribution(campaign, _epochId, _amount, currentEpoch);
+    }
+
+    function _setEpochDistribution(
```

Comment:
Set epoch distribution, what is the "distribution" being set?
Consider adding natspec comments that explain what this function is about